plotly/tools.py

```diff
@@ -4559,6 +4563,23 @@ def _scatterplot_theme(dataframe, headers, diag, size, height,
                     return fig
 
     @staticmethod
+    def _hide_tick_labels_from_box_subplots(fig):
+        """
+        Hides tick labels for box plots in scatterplotmatrix subplots.
+        """
+        boxplot_xaxes = []
+        for j, plot_data in enumerate(fig['data']):
+            if plot_data['type'] == 'box':
+                boxplot_xaxes.append(
+                    'xaxis{}'.format(plot_data['xaxis'][-1])
```

Comment:
I'm not 100% convinced this is correct. Doesn't `x1` map to `xaxis` in layout? ORRRRR, is it because in plotly.py we don't use `xaxis`, instead using `xaxis1`? Mind throwing a code comment in here?